test/resources/models/inference_example.py

```diff
@@ -0,0 +1,198 @@
+        data = _parse_csv(data)
+    else:
+        _return_error(415, 'Unsupported content type "{}"'.format(context.request_content_type or 'Unknown'))
+
+    processed_input = {
+        'headers': _make_headers(context),
+        'data': data
+    }
+
+    return processed_input
+
+
+def output_handler(data, context):
+    """Post-process TensorFlow Serving output before it is returned to the client.
+
+    Args:
+        data (obj): the TensorFlow serving response
+        context (Context): an object containing request and configuration details
+
+    Returns:
+        (dict, string): data to return to client, response content type
+    """
+    processed_output = json.loads(data.text)
+    response_content_type = context.accept_header or DEFAULT_CONTENT_TYPE
+    return processed_output, response_content_type
+
+
+def handler(data, context):
+    """Pre-process request input, and post request to TensorFlow REST API, then post-process output before it is
+    sent to the client
+
+    Args:
+        data (obj): the request data, in the format of dict or string
+        context (Context): an object containing request and configuration details
+
+    Returns:
+        (dict, string): data to return to client, response content type
+    """
+    processed_input = input_handler(data, context)
+    processed_output = _post_request(processed_input, context)
+    return str(processed_output['outputs'])
+
+
+def _post_request(data, context):
+    response = requests.post(context.rest_uri, data=data['data'], headers=data['headers'])
+
+    processed_output, _ = output_handler(response, context)
+    if 'error' in processed_output:
+        raise Exception('Invocation is not successful.')
+
+    return processed_output
+
+
+def _make_headers(context):
+    headers = {
+        'Content-Type': context.request_content_type,
+        'Accept': context.accept_header,
+        'X-Amzn-SageMaker-Custom-Attributes': _parse_custom_attributes(context)
+    }
+    return headers
+
+
+def _parse_custom_attributes(context):
+    attribute_list = []
+    if context.custom_attributes:
+        attribute_list = re.findall("(tfs-[a-z\-]+=[^,]+)", context.custom_attributes)
+    model_name = False
+    method = False
+    model_version = False
+    for attribute in attribute_list:
+        k, _ = attribute.split('=')
+        if k == 'tfs-model-name':
+            model_name = True
+        if k == 'tfs-method':
+            method = True
+        if k == 'tfs-model-version':
+            model_version = True
+
+    if not model_name:
+        attribute_list.append('tfs-model-name={}'.format(context.model_name or DEFAULT_TFS_MODEL))
+    if not method:
+        attribute_list.append('tfs-method={}'.format(context.method or DEFAULT_TFS_METHOD))
+    if not model_version and context.model_version:
+        attribute_list.append('tfs-model-version={}'.format(context.model_version))
+
+    return ','.join(attribute_list)
+
+
+def _parse_json(data):
```

Comment:
if the intention of these \_parse\_* methods is to provide a default implementation, then they should be in our service code, not in the example/test version of the code a user would supply.